src/agent/protocols/hack/update-generated-proto.sh

```diff
@@ -1,142 +1,184 @@
+proto_files_list=(
+	"agent.proto"
+	"health.proto"
+	"oci.proto"
+	"github.com/kata-containers/agent/pkg/types/types.proto"
+)
+
+# List of required command and their canonical sites.
+deps=(
+	"protoc:https://github.com/protocolbuffers/protobuf"
+	"protoc-gen-rust:https://github.com/pingcap/grpc-rs"
+	"ttrpc_rust_plugin:https://github.com/containerd/ttrpc-rust"
+)
+
+project_slug="kata-containers/kata-containers"
+project_site="github.com/${project_slug}"
+
+die()
+{
+	echo >&2 "ERROR: $*"
+	exit 1
+}
 
-die() {
-    cat <<EOT >&2
-====================================================================
-====                compile protocols failed                    ====
+info()
+{
+	echo "INFO: $*"
+}
 
-$1
+setup()
+{
+	[ -z "${GOPATH}" ] && die "GOPATH not set"
 
-====================================================================
-EOT
-    exit 1
+	local dep
+
+	for dep in "${deps[@]}"
+	do
+		local cmd=$(echo "$dep"|cut -d: -f1)
+		local url=$(echo "$dep"|cut -d: -f2-)
+
+		command -v "$cmd" &>/dev/null || die "Install $cmd from $url"
+	done
 }
 
-show_succeed_msg() {
-    echo "===================================================================="
-    echo "====                                                            ===="
-    echo "====                compile protocols succeed                   ===="
-    echo "====                                                            ===="
-    echo "===================================================================="
+usage()
+{
+	local file_list=$(echo ${proto_files_list[@]} | sed 's/ /, /g')
+
+	cat <<-EOT
+	Usage: $script_name [options] <proto-file>
+
+	Description: Generate Kata Containers agent rust API bindings
+	  from protocol buffer definition (.proto) files.
+
+	Options:
+
+	 -h : Show this help statement.
+
+	Notes:
+
+	- <proto-file> can be either:
+	  - A file (one of ${file_list}).
+	  - The word 'all', meaning process all files.
+
+EOT
 }
 
-show_usage() {
-    echo "===================================================================="
-    echo ""
-    echo "       USAGE: make PROTO_FILE=<xyz.proto> generate-protocols"
-    echo ""
-    echo "       Where PROTO_FILE may be:"
-    echo "         all: will compile all protocol buffer files"
-    echo ""
-    echo "       Or compile individually by using the exact proto file:"
-
-    # iterate over proto files
-    for file in "$@"
-    do
-        echo "         $file"
-    done
-
-    echo ""
-    echo "===================================================================="
+generate_rust_sources()
+{
+	local file="${1:-}"
+
+	[ -z "$file" ] && die "need file"
+
+	local filepath="${GOPATH}/src/${project_site}/src/agent/protocols/protos/${file}"
+
+	[ -e "$filepath" ] || die "file $file ($filepath) does not exist"
+
+	local out_dir="./protocols/src"
+	local out_file=$(basename "${file/.proto/.rs}")
+	local outfile_path="${out_dir}/${out_file}"
+
+	local ttrpc_cmd=$(command -v ttrpc_rust_plugin)
+
+	local includes=()
```

Comment:
I just noticed some remaining 1.x URLs here. ooi, I commented them out and *all* of them are redundant! I've left in only the essential includes.

Branch updated :smile: